File: Solutions/53- Maximum Subarray/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSub = nums[0];
        int currentSub = 0;

        for (var num : nums) {
            if (currentSub < 0)
                currentSub = 0;
            currentSub += num;
            maxSub = Math.max(maxSub, currentSub);
        }

        return maxSub;
    }
}

File: Solutions/53- Maximum Subarray/Solution.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        maxSub = nums[0]
        currentSub = 0

        for num in nums:
            if currentSub < 0:
                currentSub = 0
            currentSub += num
            maxSub = max(currentSub, maxSub)

        return maxSub